### PR TITLE
Fix error creating mint policy hash

### PR DESCRIPTION
**Pull request by caike**
The value returned by Blockfrost for `item.unit` for a native asset (non-Lovelace) is a:

> (...) concatenation of asset policy_id and hex-encoded asset_name

https://docs.blockfrost.io/#tag/Cardano-Transactions/paths/~1txs~1{hash}/get

`MintingPolicyHash.fromHex()` expects _just_ the policy id. This fix extracts the policy id from `item.unit`.

**Question**:

I am also trying to address the `TODO` right below this line and extract the `tokenName` from `item.unit` so it can be properly passed to the `Assets` constructor in the section that looks like this:

```javascript
/** @type {number[]} */
let token = [];

value = value.add(new Value(0n, new Assets([
  [mph, [
    [token, qty]
  ]]
])));
```

However, I am not entirely sure how to go from the hex-encoded asset name present in `item.unit` to the `number[]` type. According to this section from [the docs](https://www.hyperion-bt.org/Helios-Book/api/reference/assets.html#mintingpolicies), it's supposed to be "token name bytes" but I am not sure how to get there. Any help would be greatly appreciated.

### CODE REVIEW

File: network.js [hunk 1]
```diff
@@ -18,7 +18,8 @@ function blockFrostAmountToValue(obj) {
         if (item.unit == "lovelace") {
             value = value.add(new Value(qty));
         } else {
-            let mph = MintingPolicyHash.fromHex(item.unit);
+            let policyID = `${item.unit}`.substring(0, 56);
```
Comment (christianschmitz) [inline network.js]: if `item.unit` is a string, would it be better to write `item.unit.substring(0, 56)` directly?

File: network.js [hunk 2]
```diff
@@ -18,7 +18,8 @@ function blockFrostAmountToValue(obj) {
...
+            let mph = MintingPolicyHash.fromHex(policyID);
 
             /** @type {number[]} */
             let token = [];
```
Comment (christianschmitz) [inline network.js]: you could write `let token = hexToBytes(item.unit.substring(56))` (I think that at the top of `network.js`, `hexToBytes` is imported from the Helios library)

**Review by christianschmitz — CHANGES_REQUESTED** [review]
Thanks for figuring this out. This saves me some time

File: network.js (hunk 2 above)
Comment (caike) [reply network.js]: Works. Thanks!

File: network.js (hunk 1 above)
Comment (caike) [reply network.js]: My intention was to be overly cautious given the string data type returned by `item.unit` is not enforced by a type checker. Not opposed to calling `item.unit.substring` for simplicity though, so went ahead and made the suggested change 👍

**Comment by caike** [comment]
Added the favicon below to get rid of the console error:

PicoSwap

**Review by christianschmitz — APPROVED** [review]
Thanks for your contribution!
And thanks for taking the initiative to create the 'P' icon!

If you feel PicoSwap is a nice addition to the Cardano ecosystem we can work on a mainnet launch (is currently just a demo because this isn't my priority, it is also very difficult to get a Dapp like this marketed)

**Comment by caike** [comment]
> Thanks for your contribution! And thanks for taking the initiative to create the 'P' icon!

My pleasure. And thank you for all your work on this!

> If you feel PicoSwap is a nice addition to the Cardano ecosystem we can work on a mainnet launch (is currently just a demo because this isn't my priority, it is also very difficult to get a Dapp like this marketed)

Happy to move this project towards that goal as I learn more about Plutus and Helios. Even as it stands, just on testnet, I believe it's already a tremendously valuable learning resource to help onboard more people into Plutus 👏

**Pull request merged**